extensions/skyportal/docker-compose.skyportal.yaml

```diff
@@ -7,17 +7,65 @@ networks:
   default:
 
 services:
+  traefik:
+    image: "traefik:v2.2"
+    container_name: "traefik"
+    command:
+      # fixme: comment out if do not want the traefik dashboard on port 8090
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.web.address=:80"
+      - "--entrypoints.websecure.address=:443"
+      - "--certificatesresolvers.myresolver.acme.httpchallenge=true"
+      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
+      #- "--certificatesresolvers.myresolver.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory"
+      - "--certificatesresolvers.myresolver.acme.email=private@caltech.edu"
+      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
+    ports:
+      - "80:80"
+      - "443:443"
+      # fixme: traefik dashboard
+      - "8090:8080"
+    volumes:
+      # Create a letsencrypt dir within the folder where the docker-compose file is
+      - "../letsencrypt:/letsencrypt"
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
+
   web:
     image: skyportal/web
-    ports:
-      - "9000:5000"
-      - "9001:5001"
+    expose:
+      - "5000"
+#    ports:
+#      - "9000:5000"
     volumes:
       # fritz takes care of making a docker-appropriate config.yaml with Fritz-specific additions
       - ${PWD}/skyportal/config.yaml:/skyportal/config.yaml
+      - ${PWD}/skyportal/data/db_fritz.yaml:/skyportal/data/db_seed.yaml
+      - thumbnails:/skyportal/static/thumbnails
     command: >
       bash -c "source /skyportal_env/bin/activate
-      && pip install -r requirements.fritz.txt && (make log &) && make run"
+      && cat js-requirements.fritz.txt | xargs npm install
+      && pip install -r requirements.fritz.txt
+      && (make log &) && make run_production"
+#      && (make log &) && make run"
+    labels:
+#       Explicitly tell Traefik to expose this container
+      - "traefik.enable=true"
+      - "traefik.http.routers.fritz.rule=Host(`private.caltech.edu`)"
```

Comment:
I don't know much about traefik yet so this may be a dumb question, but would having `private.caltech.edu` explicitly put here as Host break anything with regards to deploying on another machine?

Reply:
Yea, thanks for noting -- will move it to the config. Will also make spinning traefik configurable; leaving labels here should be fine, `docker-compose` should just ignore those

Reply:
Traefik usage is now configurable.